Buildings/Fluid/Storage/Plant/BaseClasses/ReversiblePumpValveControl.mo

```diff
@@ -0,0 +1,231 @@
+within Buildings.Fluid.Storage.Plant.BaseClasses;
+block ReversiblePumpValveControl
+  "Control block for the secondary pump-valve group"
+  extends Modelica.Blocks.Icons.Block;
+
+  Modelica.Blocks.Sources.Constant conOne(k=1) "Constant y = 1"
+    annotation (Placement(transformation(extent={{-100,-90},{-80,-70}})));
+  Modelica.Blocks.Sources.Constant conZero(k=0) "Constant y = 0"
+    annotation (Placement(transformation(extent={{-100,-120},{-80,-100}})));
+  Buildings.Controls.Continuous.LimPID conPI_pum2(
+    controllerType=Modelica.Blocks.Types.SimpleController.PI,
+    Td=1,
+    k=5,
+    Ti=50)   "PI controller for pum2" annotation (Placement(transformation(
+        extent={{-10,10},{10,-10}},
+        rotation=270,
+        origin={-70,70})));
+  Modelica.Blocks.Math.Gain gaiPum2(k=1) "Gain" annotation (Placement(
+        transformation(
+        extent={{10,-10},{-10,10}},
+        rotation=90,
+        origin={-70,30})));
+  Buildings.Controls.Continuous.LimPID conPI_val2(
+    controllerType=Modelica.Blocks.Types.SimpleController.PI,
+    Td=1,
+    k=5,
+    Ti=50,
+    reverseActing=false)
+             "PI controller for val2" annotation (Placement(transformation(
+        extent={{-10,10},{10,-10}},
+        rotation=270,
+        origin={-10,68})));
+  Modelica.Blocks.Math.Gain gaiVal2(k=1) "Gain" annotation (Placement(
+        transformation(
+        extent={{10,-10},{-10,10}},
+        rotation=90,
+        origin={-10,30})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput booFloDir
+    "Flow direction: true = normal; false = reverse" annotation (Placement(
+        transformation(extent={{-120,-10},{-100,10}}), iconTransformation(
+          extent={{-140,-60},{-100,-20}})));
+  Modelica.Blocks.Interfaces.RealInput us_mTan_flow
+    "Tank mass flow rate setpoint"   annotation (Placement(transformation(
+          extent={{-10,-10},{10,10}},
+          rotation=-90,
+          origin={-70,110}), iconTransformation(
+          extent={{-10,-10},{10,10}},
+          rotation=0,
+          origin={-110,40})));
+  Modelica.Blocks.Interfaces.RealInput um_mTan_flow
+      "Measured tank mass flow rate" annotation (Placement(transformation(
+          extent={{-10,-10},{10,10}},
+          rotation=-90,
+          origin={10,110}),  iconTransformation(
+          extent={{-10,-10},{10,10}},
+          rotation=0,
+          origin={-110,80})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput booOnOff
+    "Plant output: true = on; false = off"       annotation (Placement(
```

Comment:
This signal indicates whether the plant is online (hydraulically connected to the network). When it is online, it can be outputting CHW to the network or be charged remotely by the network. When it is offline, it cannot interact with the network, but it can charge itself locally.
2be027b renamed this signal and ddf6b7c improved its description.